WalletWasabi.Fluent/ViewModels/Dialogs/Authorization/HardwareWalletAuthDialogViewModel.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using WalletWasabi.CoinJoin.Client.Clients.Queuing;
+using WalletWasabi.Fluent.Model;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Hwi;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Dialogs.Authorization
+{
+	[NavigationMetaData(Title = "Hardware Wallet")]
+	public partial class HardwareWalletAuthDialogViewModel : AuthorizationDialogBase
+	{
+		private readonly Wallet _wallet;
+		private readonly TransactionAuthorizationInfo _transactionAuthorizationInfo;
+
+		public HardwareWalletAuthDialogViewModel(Wallet wallet, TransactionAuthorizationInfo transactionAuthorizationInfo)
+		{
+			_wallet = wallet;
+			_transactionAuthorizationInfo = transactionAuthorizationInfo;
+			IsHardwareWallet = _wallet.KeyManager.IsHardwareWallet;
+			WalletIcon = _wallet.KeyManager.Icon;
+		}
+
+		public string? WalletIcon { get; }
+
+		public bool IsHardwareWallet { get; }
+
+		protected override async Task Authorize()
+		{
+			// Dequeue any coin-joining coins.
+			await _wallet.ChaumianClient.DequeueAllCoinsFromMixAsync(DequeueReason.TransactionBuilding);
```

Comment:
Dequeue is not specific to authorisation, can this be done before showing the dialog?

Reply:
ohh god... its a bug. good spot